tools/kwiver_tool_runner.cxx

```diff
@@ -22,8 +21,93 @@
 #include <memory>
 #include <utility>
 
-using applet_factory = kwiver::vital::implementation_factory_by_name< kwiver::tools::kwiver_applet >;
 using applet_context_t = std::shared_ptr< kwiver::tools::applet_context >;
+namespace kv = kwiver::vital;
+
+class command_node;
+using command_node_sptr = std::shared_ptr<command_node>;
+bool compare_node( command_node_sptr const& n1, command_node_sptr const& n2 );
+
+// -------------------------------------------------------------------
+class command_node
+{
+public:
+  command_node(std::string const& name)
+    : m_name( name ),
+      m_is_command( false )
+  { }
+
+  virtual ~command_node() = default;
+
+  void
+  add_factory( kv::plugin_factory_handle_t fact )
+  {
+    m_factory = fact;
+    m_is_command = true;
+  }
+
+  bool
+  is_command() const { return m_is_command; }
+
+  command_node_sptr
+  find( std::string const& name )
+  {
+    for ( auto& n : m_nodes )
+    {
+      if ( name == n->m_name )
+      {
+        return n;
+      }
+    } // end for
+    return nullptr;
+  }
+
+  void
+  add( command_node_sptr n )
+  {
+    m_nodes.push_back(n);
+
+    // sort vector
+    std::sort( m_nodes.begin(), m_nodes.end(), compare_node );
+    return;
+  }
+
+  void
+  print_subtree_help( std::string const& pfx = "")
+  {
+    for ( auto const& n : m_nodes )
+    {
+      n->print_subtree_help( pfx + " " + this->m_name );
+    }
+
+    if( this->is_command() )
+    {
+      std::cout << pfx << " " << this->m_name << " " << std::endl;
+      return;
+    }
+  }
+
+  void
+  dump_tree(std::string const& pfx = "") const
```

Comment:
is this just for debugging?  It doesn't look to be used and is similar to `print_subtree_help`

Reply:
It is just used for debugging the tree construction.